ext/html_tokenizer_ext/parser.c

```diff
@@ -543,15 +552,16 @@ static VALUE parser_append_data(VALUE self, VALUE source, int is_placeholder)
     parser_adjust_line_number(parser, cursor, length);
   }
   else {
+    parser_free_scan_string(parser);
```

Comment:
I'm sure there is a better way to do this but... this method can be called multiple time with more data, so this needs to be called before overwriting `parser->tk.scan.string` with a new value.